src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -313,15 +313,16 @@ private int ParseMultiSpanHeader(TRequestHandler handler, ref SequenceReader<byt
         }
 
         SequencePosition lineEnd;
-        ReadOnlySpan<byte> headerSpan;
+        scoped ReadOnlySpan<byte> headerSpan;
         if (currentSlice.Slice(reader.Position, lineEndPosition.Value).Length == currentSlice.Length - 1)
         {
             // No enough data, so CRLF can't currently be there.
             // However, we need to check the found char is CR and not LF
 
             // Advance 1 to include CR/LF in lineEnd
             lineEnd = currentSlice.GetPosition(1, lineEndPosition.Value);
-            headerSpan = currentSlice.Slice(reader.Position, lineEnd).ToSpan();
+            var header = currentSlice.Slice(reader.Position, lineEnd);
+            headerSpan = header.IsSingleSegment ? header.FirstSpan : header.ToArray();
```

Comment:
FYI @Tratcher @halter73 the new SDK version was complaining about span errors, I think related to the new compiler changes. I am not sure if there is something we can do in the `ToSpan()` extension method to address it, so I inlined the implementation here instead.

![image](https://user-images.githubusercontent.com/6995051/177300913-0650963d-2d02-4409-b1b9-2fcf119ed18c.png)

Reply:
@BrennanConroy Yep, I was seeing the errors in the other PR

Reply:
Wasn't sure if I wasn't getting the concept/syntax right

Reply:
Hmm I wonder if this could also explain: https://github.com/dotnet/aspnetcore/pull/42530#issuecomment-1172657633

https://dev.azure.com/dnceng/public/_build/results?buildId=1860786&view=ms.vss-test-web.build-test-results-tab&runId=48887712&resultId=120734&paneView=debug

which is currently blocking the runtime insertion.